lib/menus/components/radio_mode_menu.dart

```diff
@@ -73,13 +73,13 @@ Future<void> showRadioMenu(
                 if (currentRadioMode != radioModeOption) {
                   FinampSetters.setRadioMode(radioModeOption);
                 }
-                // clear tracks after updating mode to ensure any later request for radio tracks use the correct settings
-                if (radioTracksWillChange) {
-                  await queueService.clearRadioTracks();
-                }
                 if (seedItem != null && startNewQueue) {
                   unawaited(startRadioPlayback(seedItem));
                 } else {
+                  // clear tracks after updating mode to ensure any later request for radio tracks use the correct settings
+                  if (radioTracksWillChange) {
+                    await clearRadioTracks();
+                  }
```

Comment:
I'm guessing `startRadioPlayback` will invalidate the old cache state, and the new queue won't include the old radio tracks to begin with?